File: Stack.rb
class Stack
  def initialize
    @store = Array.new
  end

  def push(element)
    @store.push(element) # @store << element
  end

  def pop
    return @store.pop
  end

  def top
    return @store.last
  end

  def size
    return @store.length
  end

  def empty?
    return size == 0 # return a boolean # @store.empty? # @store.length == 0
  end

  def print
    puts "People hired:"
    puts @store
  end
end

File: stacks_queues/stack.py
from stacks_queues.linked_list import LinkedList

class StackEmptyException(Exception):
    pass

class Stack:

    def __init__(self):
        self.store = LinkedList()

    def push(self, element):
        """ Adds an element to the top of the Stack.
            Returns None
        """
        self.store.add_first(element)
       
    def pop(self):
        """ Removes an element from the top
            Of the Stack
            Raises a StackEmptyException if
            The Stack is empty.
            returns None
        """
    
        return self.store.remove_first()
       

    def empty(self):
        """ Returns True if the Stack is empty
            And False otherwise
        """
        return self.store.empty()

    def __str__(self):
        """ Returns the Stack in String form like:
            [3, 4, 7]
            Starting with the top of the Stack and
            ending with the bottom of the Stack.
        """
        current = self.store.head
        storeStacks = []
        while current:
            storeStacks.append(str(current.value))
            current = current.next
        return ", ".join(storeStacks)

